src/Source/DoctrineODMQuerySourceIterator.php

```diff
@@ -64,7 +70,23 @@ public function __construct(Query $query, array $fields, $dateTimeFormat = 'r')
             }
         }
 
-        $this->dateTimeFormat = $dateTimeFormat;
+        $formatters = array(new ArrayFormatter(), new DateTimeFormatter($dateTimeFormat), new ObjectToStringFormatter());
```

Comment:
Please put this in an else clause and trigger a deprecation error, and ask people to remove this on next major